android-pdf-viewer/src/main/java/com/github/barteksc/pdfviewer/annotation/core/pdf/PdfUtil.kt

```diff
@@ -207,19 +207,20 @@ object PdfUtil {
                 val documentId: PdfString? =
                     documentationDict.getAsString(PdfName("documentId"))
 
-                if (schemaId != null && documentId != null) {
+                if (schemaId != null) {
                     documentations.add(
                         Documentation(
                             schemaId.intValue().toLong(),
-                            documentId.toString()
+                            documentId.toString().takeIf { it.isNotEmpty() }
                         )
                     )
                 }
             }
         } else {
             return null
         }
-        return Relations(documentations)
+        val uniqueDocumentations = documentations.toSet()
+        return Relations(uniqueDocumentations.toList())
```

Comment:
Can you explain why we need to search for uniqueness now ( .toSet() ), did we introduce a bug somehow?

Reply:
The problem I see is sometimes documentation dialog is not shown upon clicking on annotation on the pdf view

I noticed this happening when the annotation has 2 same documentations, that's why I made this change in the code

This is the last executed code from `onTap`

<img width="678" alt="Screenshot 2024-08-01 at 16 47 30" src="https://github.com/user-attachments/assets/c5c640dd-37c2-4e08-9a75-badb2d27d5e8">

Tried again without doing the `.toSet()` change and the dialog was shown so having 2 same documentations is not the problem, reverted the `.toSet()` change